### PR TITLE
esp32_ble: Consider ESP_BT_STATUS_DONE a successful state

### PR DESCRIPTION
**Pull request by polyfloyd**
# What does this implement/fix?

This fixes the log messages below from repeatedly being written:
```
[00:23:57][E][esp32_ble_tracker:194]: Scan set param failed: 5
[00:23:57][D][esp32_ble_tracker:266]: Starting scan...
[00:23:57][D][esp32_ble_tracker:186]: Stopping scan after failure...
```

The issue seems to arise from BT_STATE_DONE being interpreted as an error. Following the documentation for this value leads to `packages/framework-espidf/components/bt/common/include/bt_common.h` describing it as `request already completed`. Assuming that whatever was being performed is idempotent, mapping this state to success should be ok.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

Related issue: https://github.com/esphome/issues/issues/4058

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
esp32_ble_tracker:
  id: ble_tracker
  scan_parameters:
    interval: 20s
    window: 10s
    duration: 1min
    active: false
    continuous: true

sensor:
  - platform: xiaomi_hhccjcy01
    mac_address: 'FF:FF:FF:FF:FF:FF'
    temperature:
      name: "Miflora 01 Temperature"
    moisture:
      name: "Miflora 01 Moisture"
    illuminance:
      name: "Miflora 01 Illuminance"
    conductivity:
      name: "Miflora 01 Soil Conductivity"
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

### CODE REVIEW

**Comment by polyfloyd** [comment]
Sooo, anything else that needs to happen? I consider this ready to be reviewed

**Comment by polyfloyd** [comment]
@clydebarrow Could you please have a glance?

File: esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp [hunk 1]
```diff
@@ -364,6 +364,10 @@ void ESP32BLETracker::gap_event_handler(esp_gap_ble_cb_event_t event, esp_ble_ga
 }
 
 void ESP32BLETracker::gap_scan_set_param_complete_(const esp_ble_gap_cb_param_t::ble_scan_param_cmpl_evt_param &param) {
+  if (param.status == ESP_BT_STATUS_DONE) {
+    this->scan_set_param_failed_ = ESP_BT_STATUS_SUCCESS;
+    return;
+  }
   this->scan_set_param_failed_ = param.status;
```
Comment (clydebarrow) [inline esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp]:
```suggestion
  if (param.status == ESP_BT_STATUS_DONE) {
    this->scan_set_param_failed_ = ESP_BT_STATUS_SUCCESS;
  } else {
    this->scan_set_param_failed_ = param.status;
  }
```

Comment (clydebarrow) [reply esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp]: Not a big deal, but in general it's best to avoid an embedded `return` if possible.

**Review by clydebarrow — APPROVED** [review]
LGTM

**Comment by polyfloyd** [comment]
Thanks!

**Pull request merged**